crates/phactory/src/side_task.rs

```diff
@@ -1,21 +1,31 @@
 use crate::storage::Storage;
 use ::chain::BlockNumber;
-use phala_mq::MessageSendQueue;
+use phala_mq::{MessageSendQueue, Sr25519Signer};
+use serde::{Deserialize, Serialize};
 
-type SigningMessage = phala_mq::SigningMessage<sp_core::sr25519::Pair>;
+type SigningMessage = phala_mq::SigningMessage<Sr25519Signer>;
 
 pub struct PollContext<'a> {
     pub block_number: BlockNumber,
     pub send_mq: &'a MessageSendQueue,
     pub storage: &'a Storage,
 }
 
+type OnFinish = Box<dyn FnOnce(&PollContext) -> Option<Vec<SigningMessage>> + Send + 'static>;
+
+#[derive(Serialize, Deserialize)]
 struct TaskWrapper {
-    on_finish: Box<dyn FnOnce(&PollContext) -> Option<Vec<SigningMessage>> + Send + 'static>,
+    #[serde(skip)]
+    #[serde(default = "zombie")]
+    on_finish: OnFinish,
     default_messages: Vec<SigningMessage>,
     end_block: BlockNumber,
 }
 
+fn zombie() -> OnFinish {
```

Comment:
Yes, the already arrived but not yet committed result will be dropped if restart.

Reply:
In this way, it will try to submit an empty message to the blockchain, right?

Reply:
Btw, it sounds like we need to document it in the code somewhere

Reply:
> In this way, it will try to submit an empty message to the blockchain, right?
The default_messages if has any will be submitted.
Added some comments.